Reconstruction/RecCalorimeter/src/components/SingleCaloTowerTool.h

```diff
@@ -0,0 +1,127 @@
+#ifndef RECCALORIMETER_SINGLECALOTOWERTOOL_H
+#define RECCALORIMETER_SINGLECALOTOWERTOOL_H
+
+// from Gaudi
+#include "GaudiAlg/GaudiTool.h"
+
+// FCCSW
+#include "FWCore/DataHandle.h"
+#include "RecInterface/ITowerTool.h"
+#include "DetSegmentation/GridPhiEta.h"
+class IGeoSvc;
+
+// datamodel
+namespace fcc {
+class CaloHitCollection;
+}
+
+namespace DD4hep {
+namespace DDSegmentation {
+class Segmentation;
+}
+}
+
+/** @class SingleCaloTowerTool Reconstruction/RecCalorimeter/src/components/SingleCaloTowerTool.h SingleCaloTowerTool.h
+ *
+ *  Tool building the calorimeter towers for the sliding window algorithm.
+ *  Towers are built of cells in eta-phi, summed over all radial layers.
+ *  A tower contains all cells within certain eta and phi (tower size: '\b deltaEtaTower', '\b deltaPhiTower').
+ *  Distance in r plays no role, however `\b radiusForPosition` needs to be defined
+ *  (e.g. to inner radius of the detector) for the cluster position calculation. By default the radius is equal to 1.
+ *
+ *  This tool creates towers from a single cell collection (from one calorimeter).
+ *
+ *  For more explanation please [see reconstruction documentation](@ref md_reconstruction_doc_reccalorimeter).
+ *
+ *  @author Anna Zaborowska
+ *  @author Jana Faltova
+ */
+
+class SingleCaloTowerTool : public GaudiTool, virtual public ITowerTool {
+public:
+  SingleCaloTowerTool(const std::string& type, const std::string& name, const IInterface* parent);
+  virtual ~SingleCaloTowerTool() = default;
+  /**  Initialize.
+   *   @return status code
+   */
+  virtual StatusCode initialize() final;
+  /**  Finalize.
+   *   @return status code
+   */
+  virtual StatusCode finalize() final;
+  /**  Find number of calorimeter towers.
+   *   Number of towers in phi is calculated from full azimuthal angle (2 pi) and the size of tower in phi ('\b deltaPhiTower').
+   *   Number of towers in eta is calculated from maximum detector eta ('\b etaMax`) and the size of tower in eta ('\b deltaEtaTower').
+   *   If max eta is undefined, 0 is returned. In this case number of eta towers should be calculated for each event separately (etaTowersNumber()).
+   *   @return Array containing number of towers in eta and phi.
+   */
+  virtual std::array<int,2> towersNumber() final;
+  /**  Find number of calorimeter towers in eta for current event.
+   *   Cell collection is searched for a highest eta (absolute value).
+   *   @return Number of towers in eta for current event.
+   */
+  virtual int etaTowersNumber() final;
+  /**  Build calorimeter towers.
+   *   Tower is segmented in eta and phi, with the energy from all layers (no r segmentation).
+   *   Currently the size of tower needs to be a multiple of a cell size (so each cell belongs to only one tower).
+   *   TODO: split cell energy into more towers if cell size is larger than the tower.
```

Comment:
Please move the to-do items out of the code to the issue tracker (or the one associated to your own repository, if you prefer not to add issues that are associated with potentially WIP code)